### DISCUSSION TITLE
[bug]: Why do I write this animation doesn't perform properly?

### DISCUSSION THREAD

**Discussion by Deja-vuuu** [pmndrs/react-spring#1922] [Support]
### Which react-spring target are you using?

- [X] `@react-spring/web`
- [ ] `@react-spring/three`
- [ ] `@react-spring/native`
- [ ] `@react-spring/konva`
- [ ] `@react-spring/zdog`

### What version of react-spring are you using?

0.7.8

### What's Wrong?

Why do I write this animation doesn't perform properly?
I'm a new student. Thank you very much。

### To Reproduce

[Why do I write this animation doesn't perform properly?](https://stackoverflow.com/questions/72740685/why-do-i-write-this-animation-doesnt-perform-properly)

### Expected Behaviour

Animation executes normally after display

### Link to repo

https://stackoverflow.com/questions/72740685/why-do-i-write-this-animation-doesnt-perform-properly

**Comment by joshuaellis** [comment]
Your spring will run on component mount so when you change the state the spring has already updated. To be honest for this use case, you're probably better off using `useTransition`.
